docs/madr/decisions/057-sni-in-the-resource.md

```diff
@@ -0,0 +1,130 @@
+# SNI in the resource definition
+
+* Status: accepted
+
+Technical Story: 
+
+## Context and Problem Statement
+
+SNI is used for cross-zone communication to proxy the traffic through zone proxies.
+Recently we decided on SNI format in MADR 053.
+
+The SNI is constructed on the server and the client side. This approach has a couple of downsides
+* We hit multiple edge cases of constructing it on the client side, because of resource hashing.
+  This logic is not as simple as it should be.
+* Any migration of this format would be hard to execute
+* If we introduce a functionality to enter the mesh directly through zone ingress (assuming that cert is acquired) it would be hard for the client to construct this.
+
+## Considered Options
+
+* SNI in the resource definition
+* Keep it implicit
+
+## Decision Outcome
+
+Chosen option: "Option 1 - SNI in the resource definition", because it solves the problems described in previous section.
+For `MeshMultiZoneService` "Option 2 - compute ports on global cp", because advantages outweights other option.
+
+## Pros and Cons of the Options
+
+### Option 1 - SNI in the resource definition
+
+SNI is unique for each port, therefore we need to define it for each port
+
+```yaml
+kind: MeshService
+spec:
+  ports:
+    - port: 80
+      targetPort: 8080
+      sni: ae10a8071b8a8eeb8.backend.8080.demo.ms # new field
```

Comment:
I guess migration is still not possible like this (we'd need to have possibility to have multiple SNIs to support migration). But I guess this is a start at least.

If we wanted to support migration I guess we would need:

```yaml
sni:
   server: [a, b] # zoneIngress
   client: a
```

So migration would happen:

1. Migrate all servers to support both
```yaml
sni:
   server: [old, new]
   client: old
```

2. Migrate clients to new one:
```yaml
sni:
   server: [old, new]
   client: new
```

3. Remove old from server:
```yaml
sni:
  server: [new]
  client: new
```

For sure this is too cumbersome to expose from the start. But should we at least make the model flexible for the future?

.e.g:

```yaml
sni:
  client: new
```

Maybe client is not the right field name. But the ability to extend SNI in the future seems required

Reply:
Ha or by migration you mean we wouldn't update existing SNIs?

Reply:
You are right. Instead of this server client etc, we could do this
```
snis:
- new
- old
```
and just recommend the client to always use the first one on the list

Reply:
Please remember to update the madr with this new `snis` array